docs/development/new-packages.md

```diff
@@ -96,43 +96,72 @@ then restart the build.
+
+1. Clone the git repository of the package. You might want to use the options `git clone --depth 1 --branch <version-tag>`.
+2. Make a new branch with `git checkout -b pyodide-old-version` (e.g., `pyodide-1.21.4`).
+3. Apply the current patches with `git am <pyodide-root>/packages/<package-name>/patches/*`.
+4. Make a new branch `git checkout -b pyodide-new-version` (e.g., `pyodide-1.22.0`)
+5. Rebase the patches with `git rebase old-version --onto new-version`
+   (e.g., `git rebase pyodide-1.21.4 --onto pyodide-1.22.0`). Resolve any rebase
+   conflicts. If a patch has been upstreamed, you can drop it with `git rebase --skip`.
+6. Remove old patches with `rm <pyodide-root>/packages/<package-name>/patches/*`.
+7. Use `git format-patch <version-tag> -o <pyodide-root>/packages/<package-name>/patches/`
+   to generate new patch files.
+
+#### Upstream your patches!
+
+Please create PRs or issues to discuss with the package to try to find ways to
+include your patches into the package. Many package authors are very receptive
+to including Pyodide-related patches and they reduce future maintenance work for
+us.
 
 ## The package build pipeline
 
-Pyodide includes a toolchain to make it easier to add new third-party Python
-libraries to the build. We automate the following steps:
-
-- Download a source tarball (usually from PyPI)
-- Confirm integrity of the package by comparing it to a checksum
-- Apply patches, if any, to the source distribution
-- Add extra files, if any, to the source distribution
-- If the package includes C/C++/Cython extensions:
-  - Build the package natively, keeping track of invocations of the native
-    compiler and linker
-  - Rebuild the package using emscripten to target WebAssembly
-- If the package is pure Python:
-  - Run the `setup.py` script to get the built package
-- Unvendor unit tests included in the installation folder to a separate package
-  `<package name>-tests`
-- Package the results into an emscripten virtual filesystem package, which
-  comprises:
-  - A `.data` file containing the file contents of the whole package,
-    concatenated together
-  - A `.js` file which contains metadata about the files and installs them into
-    the virtual filesystem.
-
-Lastly, a `packages.json` file is output containing the dependency tree of all
+Pyodide includes a toolchain to add new third-party Python libraries to the
+build. We automate the following steps:
+
+- If source is a url (not in-tree):
+  - Download a source archive or a pure python wheel (usually from PyPI)
+  - Confirm integrity of the package by comparing it to a checksum
+  - If building from source (not from a wheel):
+    - Apply patches, if any, to the source distribution
+    - Add extra files, if any, to the source distribution
+- If the source is not a wheel (building from a source archive or an in-tree
+  source):
+  - Run `build/script` if present
+  - Install wrappers for `gfortran`, `gcc`, `g++`, `ar`, and `ld` to intercept
```

Comment:
Maybe "Use wrappers" or "Create symlinks for wrappers of", as "install" sounds like it's going to install gfortran etc which is not the case.

Reply:
Incidentally, we could just check the symlinks in `tools/symlinks` into the repository. I decided not to do this because I think it's clearer to keep the "source of truth" for that in pywasmcross. Though we should ideally update pywasmcross to delete all the symlinks in `tools/symlinks` before creating new ones...